aten/src/ATen/native/ForeachOps.cpp

```diff
@@ -0,0 +1,26 @@
+
+#include <ATen/native/ForeachOps.h>
+#include <ATen/native/DispatchStub.h>
+
+namespace at {
+namespace native {
+
+std::vector<Tensor> _foreach_add(TensorList tensors, Scalar scalar) {
+  if (tensors.size() == 0) {
+    throw std::runtime_error("Tensor list can't be empty.");
+  }
```

Comment:
Why not? Just return an empty list in this case?

Reply:
`cat` does not support empty list

Reply:
Sure, but that's because `cat` can't be easily implemented in this place. This doesn't seem to be the case for foreach ops though?

Reply:
Update: its not that easy to do because of dispatcher fallback. Leaving as an error for now.